### PR TITLE
Catch null errors

### PR DESCRIPTION
**Pull request by core23**
## Subject

Found some null errors when running `console translation:update`

I am targeting this branch, because this is a patch.

## Changelog

```markdown
### Fixed
- Catch null errors when accessing request
- Catch null errors when loading unknown block
```

### CODE REVIEW

File: src/Admin/BaseBlockAdmin.php [hunk 1]
```diff
@@ -273,6 +273,10 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
      */
     private function loadBlockDefaults(BlockInterface $block)
     {
+        if (!$this->blockManager->has($block->getType() ?? '')) {
+            return $block;
+        }
```
Comment (VincentLanglet) [inline src/Admin/BaseBlockAdmin.php:278]:
```suggestion
        if (null === $block->getType() || !$this->blockManager->has($block->getType())) {
            return $block;
        }
```

?

File: src/Admin/BlockAdmin.php [hunk 2]
```diff
@@ -203,7 +207,12 @@ private function getDefaultTemplate(BlockServiceInterface $blockService)
 
     private function configureBlockFields(FormMapper $formMapper, BlockInterface $block): void
     {
+        if (!$this->blockManager->has($block->getType() ?? '')) {
+            return;
+        }
```
Comment (VincentLanglet) [inline src/Admin/BlockAdmin.php:212]: same

File: src/Admin/BlockAdmin.php [hunk 3]
```diff
@@ -203,7 +207,12 @@ private function getDefaultTemplate(BlockServiceInterface $blockService)
...
+
         $service = $this->blockManager->get($block);
+
         $blockType = $block->getType();
```
Comment (VincentLanglet) [inline src/Admin/BlockAdmin.php:216]: You could use this variable for the previous check

File: src/Admin/SharedBlockAdmin.php [hunk 4]
```diff
@@ -90,6 +90,10 @@ protected function configureFormFields(FormMapper $formMapper)
 
     private function configureBlockFields(FormMapper $formMapper, BlockInterface $block): void
     {
+        if (!$this->blockManager->has($block->getType() ?? '')) {
+            return;
+        }
```
Comment (VincentLanglet) [inline src/Admin/SharedBlockAdmin.php:95]: same

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: src/Admin/BaseBlockAdmin.php [hunk 5]
```diff
@@ -273,6 +273,10 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
      */
     private function loadBlockDefaults(BlockInterface $block)
     {
+        if (null === $this->blockManager->has($block->getType() ?? '')) {
+            return $block;
+        }
```
Comment (VincentLanglet) [inline src/Admin/BaseBlockAdmin.php:278]: null === $block->getType() || !$this->blockManager->has($block->getType())

I think you missed your changes

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: src/Admin/BaseBlockAdmin.php [hunk 6]
```diff
@@ -273,6 +273,10 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
      */
     private function loadBlockDefaults(BlockInterface $block)
     {
+        if (null === $block->getType() || null === $this->blockManager->has($block->getType())) {
+            return $block;
+        }
```
Comment (VincentLanglet) [inline src/Admin/BaseBlockAdmin.php:278]:
```suggestion
        if (null === $block->getType() || !$this->blockManager->has($block->getType())) {
            return $block;
        }
```

File: src/Admin/BlockAdmin.php [hunk 7]
```diff
@@ -203,6 +207,10 @@ private function getDefaultTemplate(BlockServiceInterface $blockService)
 
     private function configureBlockFields(FormMapper $formMapper, BlockInterface $block): void
     {
+        if (null === $block->getType() || null === $this->blockManager->has($block->getType())) {
+            return;
+        }
```
Comment (VincentLanglet) [inline src/Admin/BlockAdmin.php:212]:
```suggestion
        if (null === $block->getType() || !$this->blockManager->has($block->getType())) {
            return;
        }
```

File: src/Admin/SharedBlockAdmin.php [hunk 8]
```diff
@@ -90,6 +90,10 @@ protected function configureFormFields(FormMapper $formMapper)
 
     private function configureBlockFields(FormMapper $formMapper, BlockInterface $block): void
     {
+        if (null === $block->getType() || null === $this->blockManager->has($block->getType())) {
+            return;
+        }
```
Comment (VincentLanglet) [inline src/Admin/SharedBlockAdmin.php:95]:
```suggestion
        if (null === $block->getType() || !$this->blockManager->has($block->getType())) {
            return;
        }
```

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: src/Admin/BaseBlockAdmin.php [hunk 9]
```diff
@@ -273,6 +273,12 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
      */
     private function loadBlockDefaults(BlockInterface $block)
     {
+        $blockType = $block->getType();
+
+        if (null === $blockType || $this->blockManager->has($blockType)) {
```
Comment (VincentLanglet) [inline src/Admin/BaseBlockAdmin.php:278]:
```suggestion
        if (null === $blockType || !$this->blockManager->has($blockType)) {
```

File: src/Admin/BlockAdmin.php [hunk 10]
```diff
@@ -203,9 +207,14 @@ private function getDefaultTemplate(BlockServiceInterface $blockService)
 
     private function configureBlockFields(FormMapper $formMapper, BlockInterface $block): void
     {
-        $service = $this->blockManager->get($block);
         $blockType = $block->getType();
 
+        if (null === $blockType || $this->blockManager->has($blockType)) {
```
Comment (VincentLanglet) [inline src/Admin/BlockAdmin.php:212]:
```suggestion
        if (null === $blockType || !$this->blockManager->has($blockType)) {
```

File: src/Admin/SharedBlockAdmin.php [hunk 11]
```diff
@@ -90,12 +90,18 @@ protected function configureFormFields(FormMapper $formMapper)
 
     private function configureBlockFields(FormMapper $formMapper, BlockInterface $block): void
     {
+        $blockType = $block->getType();
+
+        if (null === $blockType || $this->blockManager->has($blockType)) {
```
Comment (VincentLanglet) [inline src/Admin/SharedBlockAdmin.php:95]:
```suggestion
        if (null === $blockType || !$this->blockManager->has($blockType)) {
```

File: src/Admin/BaseBlockAdmin.php (hunk 9 above)
Comment (core23) [reply src/Admin/BaseBlockAdmin.php:278]: Good catch!

**Review by VincentLanglet — DISMISSED** [review]

**Review by wbloszyk — DISMISSED** [review]

File: src/Admin/PageAdmin.php [hunk 12]
```diff
@@ -322,48 +328,48 @@ protected function configureFormFields(FormMapper $formMapper)
             ->end()
         ;
 
-        if (!$this->getSubject() || ($this->getSubject() && $this->getSubject()->getParent()) || ($this->getSubject() && !$this->getSubject()->getId())) {
+        if (null === $page || ($page && $page->getParent()) || ($page && !$page->getId())) {
```
Comment (phansys) [inline src/Admin/PageAdmin.php:331]:
```suggestion
        if (null === $page || ($page && $page->getParent()) || ($page && null === $page->getId())) {
```

File: src/Admin/PageAdmin.php [hunk 13]
```diff
@@ -284,15 +288,17 @@ protected function configureFormFields(FormMapper $formMapper)
...
+        if (null === $page || (!$page->isInternal() && !$page->isError())) {
             $formMapper
                 ->with('form_page.group_main_label')
                     ->add('url', TextType::class, ['attr' => ['readonly' => true]])
                 ->end()
             ;
         }
 
-        if ($this->hasSubject() && !$this->getSubject()->getId()) {
+        if (null !== $page && !$page->getId()) {
```
Comment (phansys) [inline src/Admin/PageAdmin.php:301]:
```suggestion
        if (null !== $page && null === $page->getId()) {
```

**Review by phansys — CHANGES_REQUESTED** [review]

**Review by phansys — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Comment by phansys** [comment]
Thank you @core23!

**Pull request merged**